### PR TITLE
Updated to CRA4 and React 17 (and more...)

### PR DESCRIPTION
**Pull request by albertoarana**

### CODE REVIEW

**Comment by albertoarana** [comment]
Tip: never update `react-map-gl` to version 6, use mapbox v2 instead of v1

File: template-sample-app/template/src/components/common/Map.js [hunk 1]
```diff
@@ -53,12 +52,14 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-export function Map(props) {
+function Map(props) {
```
Comment (albertoarana) [inline template-sample-app/template/src/components/common/Map.js:55]: Destructuring

File: template-sample-app/template/src/App.js [hunk 2]
```diff
@@ -63,7 +62,9 @@ function App() {
         ) : (
           <>
             <Header />
-            {routing}
+            <React.Suspense fallback={<></>}>
```
Comment (albertoarana) [inline template-sample-app/template/src/App.js:65]: Without React

File: template-sample-app/template/src/components/legends/Legend.js [hunk 3]
```diff
@@ -22,11 +22,14 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-export function Legend(props) {
+function Legend(props) {
```
Comment (albertoarana) [inline template-sample-app/template/src/components/legends/Legend.js:25]: Destructuring

File: template-sample-app/template/src/routes.js [hunk 4]
```diff
@@ -13,6 +12,8 @@ import Datasets from 'components/views/datasets/Datasets';
 import NotFound from 'components/views/NotFound';
 // Auto import
 
+const Main = React.lazy(() => import('components/views/Main'));
```
Comment (albertoarana) [inline template-sample-app/template/src/routes.js:15]: No React

File: template-skeleton/template/src/App.js [hunk 5]
```diff
@@ -63,7 +62,9 @@ function App() {
         ) : (
           <>
             <Header />
-            {routing}
+            <React.Suspense fallback={<></>}>
```
Comment (albertoarana) [inline template-skeleton/template/src/App.js:65]: No React

File: template-skeleton/template/src/components/common/Map.js [hunk 6]
```diff
@@ -53,14 +52,15 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-export function Map(props) {
+function Map(props) {
```
Comment (albertoarana) [inline template-skeleton/template/src/components/common/Map.js:55]: Destructuring

File: template-skeleton/template/src/components/views/Main.js [hunk 7]
```diff
@@ -155,16 +154,19 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-export default function Main() {
+function Main() {
+  const classes = useStyles();
   const dispatch = useDispatch();
+
+  const mobileContainer = React.useRef(null);
```
Comment (albertoarana) [inline template-skeleton/template/src/components/views/Main.js:161]: No React

File: template-skeleton/template/src/routes.js [hunk 8]
```diff
@@ -1,10 +1,11 @@
 import React from 'react';
 import { Navigate } from 'react-router-dom';
 import { OAuthCallback } from '@carto/react/oauth';
-import Main from 'components/views/Main';
 import NotFound from 'components/views/NotFound';
-
 // Auto import
+
+const Main = React.lazy(() => import('components/views/Main'));
```
Comment (albertoarana) [inline template-skeleton/template/src/routes.js:7]: No React

**Comment by VictorVelarde** [comment]
Setting this one as draft, as we discuss on each piece

**Comment by VictorVelarde** [comment]
I think we can probably close this PR now, ok with it @albertoarana ?

**Comment by albertoarana** [comment]
I agree. I close this PR

**Pull request closed without merging**